plugins/e57/io/Scan.cpp

```diff
@@ -130,70 +119,118 @@ void Scan::decodeHeader()
     // Get pose estimation
     getPose();
 
-    // Get bounds
-    for (auto& field: supportedFields)
+    // Get rescale factors for new scan, These factors are only for
+    // dimensions with limits. This is required Since,
+    // 1. E57 support colors in uint8 as well as uint16. This is specified
+    //    in header as "colorLimits".
+    //      - If colorLimits is 0-255 then it is uint8 and if 0-65535 then
+    // it
+    // is uint16.
+    //      - To make this consistant, We are rescaling colors to 0-65535
+    //        range, Since default types for colors in PDAL are Unsigned16.
+    // 2. E57 supports intensity in float ranging 0-1.  This is specified in
+    //    header as "intensityLimits".
+    //      - Since default type for intensity in PDAL is Unsigned16, E57
+    //        intensity (between 0-1) need to be rescaled in uint16 (between
+    //        0-65535).
+    // To do the rescling we need the rescale factors so that we can
+    // directly multipy them with colors and intensity values. E.g. - If
+    // color limit is 0-255 then rescale factor would be 257.00
+    //        (double value) i.e 65535/(255-0)=257.
+    //      - If color limit is 0-65535 then rescale factor would be 1.00
+    //        (double value) i.e 65535/(65535-0)= 1.
+    for (auto& field : supportedFields)
     {
-        auto minmax = pdal::e57plugin::getLimits(*m_rawData,field);
-        //ABELL - This seems weird.  Could you not have one value that's
-        //  NaN and another not NaN?
-        if (minmax == minmax) // not nan
-            m_valueBounds[pdal::e57plugin::e57ToPdal(field)] = minmax;
+        auto minmax = std::make_pair(0.0, 0.0);
+        if (pdal::e57plugin::getLimits(*m_rawData, field, minmax))
+        {
+            m_rescaleFactors[pdal::e57plugin::e57ToPdal(field)] =
+                (std::numeric_limits<uint16_t>::max)() /
+                (minmax.second - minmax.first);
```

Comment:
- For intensities the scaling would be 0->1 to 0->65535. But cannot not be same for all, Colors may need to be scaled from 0->255 to 0->65535.

- `supportedFields` is a list of E57 dimensions which are supported in PDAl.

Reply:
But this isn't what you're doing.  You're scaling the current range -- whatever that is -- to 0-65535.

Reply:
Hmm got your point.. 
I think we need to get rid of this scaling stuff. We should forward what we get from E57.
Only intensities are required to scale if they are between 0-1. Since we cannot forward them as a double value.

Reply:
Done.